consensus/types/src/light_client_header.rs

```diff
@@ -1,26 +1,183 @@
-use crate::test_utils::TestRandom;
+use crate::beacon_block_body::NUM_BEACON_BLOCK_BODY_HASH_TREE_ROOT_LEAVES;
 use crate::BeaconBlockHeader;
+use crate::{light_client_update::*, BeaconBlockBody};
+use crate::{
+    test_utils::TestRandom, EthSpec, ExecutionPayloadHeaderCapella, ExecutionPayloadHeaderDeneb,
+    FixedVector, Hash256, SignedBeaconBlock,
+};
+use merkle_proof::verify_merkle_proof;
 use serde::{Deserialize, Serialize};
 use ssz_derive::{Decode, Encode};
+use std::marker::PhantomData;
+use superstruct::superstruct;
 use test_random_derive::TestRandom;
+use tree_hash::TreeHash;
 
-#[derive(
-    Debug,
-    Clone,
-    PartialEq,
-    Serialize,
-    Deserialize,
-    Encode,
-    Decode,
-    TestRandom,
-    arbitrary::Arbitrary,
+#[superstruct(
+    variants(Altair, Capella, Deneb),
+    variant_attributes(
+        derive(
+            Default,
+            Debug,
+            Clone,
+            PartialEq,
+            Serialize,
+            Deserialize,
+            Encode,
+            Decode,
+            TestRandom,
+            arbitrary::Arbitrary,
+        ),
+        serde(bound = "E: EthSpec", deny_unknown_fields),
+        arbitrary(bound = "E: EthSpec"),
+    )
 )]
-pub struct LightClientHeader {
+#[derive(Debug, Clone, Serialize, Deserialize, Encode, Decode, arbitrary::Arbitrary, PartialEq)]
+#[serde(untagged)]
+#[serde(bound = "E: EthSpec", deny_unknown_fields)]
+#[ssz(enum_behaviour = "union")]
+#[arbitrary(bound = "E: EthSpec")]
+pub struct LightClientHeader<E: EthSpec> {
```

Comment:
I think `ForkVersionDeserialize` implemented for this struct

Reply:
added `ForkVersionDeserialize`